cdap-app-templates/cdap-etl/cdap-data-streams/src/main/java/co/cask/cdap/datastreams/SparkStreamingPipelineDriver.java

```diff
@@ -84,16 +85,19 @@ public void run(final JavaSparkExecutionContext sec) throws Exception {
     final DataStreamsPipelineSpec pipelineSpec = GSON.fromJson(sec.getSpecification().getProperty(Constants.PIPELINEID),
                                                                DataStreamsPipelineSpec.class);
 
+    Set<StageSpec> stageSpecs = pipelineSpec.getStages();
     final PipelinePhase pipelinePhase = PipelinePhase.builder(SUPPORTED_PLUGIN_TYPES)
       .addConnections(pipelineSpec.getConnections())
-      .addStages(pipelineSpec.getStages())
+      .addStages(stageSpecs)
       .build();
 
     boolean checkpointsDisabled = pipelineSpec.isCheckpointsDisabled();
+    boolean isPreviewEnabled =
+      stageSpecs.isEmpty() || sec.getDataTracer(stageSpecs.iterator().next().getName()).isEnabled();
```

Comment:
Does `stageSpecs` guaranteed to be non-empty?

Reply:
Unsure, but if it is then we shouldn't be checkpointing anyways. Did this just to be safe